File: LWLogger/Classes/LWLogUtil.m
//
//  LWLogUtil.m
//  CocoaLumberjack
//
//  Created by luowei on 2019/11/8.
//

#import "LWLogUtil.h"
#import "ZipArchive.h"

@interface LWLogUtil ()
@end

@implementation LWLogUtil
+ (void)initLogConfig {
    [DDLog addLogger:[DDOSLogger sharedInstance]];

    DDFileLogger *fileLogger = [[DDFileLogger alloc] init]; // File Logger
    fileLogger.rollingFrequency = 60 * 60 * 24; // 24 hour rolling
    fileLogger.logFileManager.maximumNumberOfLogFiles = 7;
    [DDLog addLogger:fileLogger];
}

+ (NSString *)logZipPath {
    //获取DDLog打印的日志
    DDFileLogger *fileLogger = [[DDFileLogger alloc] init];
    //获取log文件夹路径
    NSString *logDirectory = [fileLogger.logFileManager logsDirectory];
    DDLogDebug(@"%@", logDirectory);
    //zip文件路径
    NSDateFormatter *fmt = [[NSDateFormatter alloc] init];
    [fmt setDateFormat:@"yyyy-MM-dd-HH:mm:ss"];
    NSString *zipFileName = [NSString stringWithFormat:@"/feadbackLog-%@.zip", [fmt stringFromDate:[NSDate date]]];
    NSString *logZipPath = [logDirectory stringByAppendingString:zipFileName];

    //获取排序后的log名称
    NSArray <NSString *> *logsNameArray = [fileLogger.logFileManager sortedLogFileNames];
    DDLogDebug(@"%@", logsNameArray);
    [SSZipArchive createZipFileAtPath:logZipPath withFilesAtPaths:logsNameArray];

    return logZipPath;
}


@end

File: LWLogger_swift/Classes/LWLogUtil.swift
//
//  LWLogUtil.swift
//  LWLogger
//
//  Swift implementation of LWLogUtil
//  Created by luowei on 2019/11/8.
//

import Foundation
import CocoaLumberjack

/// Logging utility class that wraps CocoaLumberjack functionality
/// Provides file-based logging with rotation and zip archive creation
@objc public class LWLogUtil: NSObject {

    // MARK: - Public Methods

    /// Initializes the logging configuration
    /// Sets up console logging and file logging with 24-hour rotation
    /// Keeps maximum of 7 log files
    @objc public static func initLogConfig() {
        // Add OS Logger (console output)
        DDLog.add(DDOSLogger.sharedInstance)

        // Add File Logger
        let fileLogger = DDFileLogger()
        fileLogger.rollingFrequency = 60 * 60 * 24 // 24 hour rolling
        fileLogger.logFileManager.maximumNumberOfLogFiles = 7
        DDLog.add(fileLogger)
    }

    /// Creates a zip archive of all log files
    /// - Returns: Path to the created zip file
    @objc public static func logZipPath() -> String {
        // Get DDLog log files
        let fileLogger = DDFileLogger()

        // Get log folder path
        let logDirectory = fileLogger.logFileManager.logsDirectory
        DDLogDebug(logDirectory)

        // Create zip file path
        let dateFormatter = DateFormatter()
        dateFormatter.dateFormat = "yyyy-MM-dd-HH:mm:ss"
        let zipFileName = "/feedbackLog-\(dateFormatter.string(from: Date())).zip"
        let logZipPath = logDirectory + zipFileName

        // Get sorted log file names
        let logsNameArray = fileLogger.logFileManager.sortedLogFileNames
        DDLogDebug("\(logsNameArray)")

        // Create zip archive
        SSZipArchive.createZipFile(atPath: logZipPath, withFilesAtPaths: logsNameArray)

        return logZipPath
    }
}
